### ISSUE TITLE
Carthage build failed on Xcode 12

### ISSUE THREAD

**Issue by denis-ionov** [microsoft/appcenter-sdk-apple#2168]
Carthage build failed on building scheme "AppCenterCrashes macOS Framework". Task failed with exit code 65.

**Comment by Jamminroot** [comment]
Hello, @denis-ionov, thanks for reaching out.

We've tested `carthage update`, `carthage build`, project builds, launches and user is out there in overview in App Center portal.

Version's used:
XCode 12
iOS 14
Carthage 0.36.0

Can you kindly elaborate? Preferrably using a [template](https://github.com/microsoft/appcenter-sdk-apple/blob/develop/.github/ISSUE_TEMPLATE/problem_report.md)? That would really help us troubleshoot the problem with the SDK.

**Comment by denis-ionov** [comment]
@Jamminroot 
**Command** that i use:
`XCODE_XCCONFIG_FILE="path/to/Carthage.xcconfig" carthage build --no-use-binaries --platform macOS`

**xcconfig file**
```
VALID_ARCHS = x86_64;
ONLY_ACTIVE_ARCH = YES;
MACOSX_DEPLOYMENT_TARGET = 10.12;
SWIFT_VERSION = 5.0;
```

**Versions**
Xcode: 12.0 (12A7209)
macOS: 10.15.6 (19G2021)
Carthage: 0.36.0

With **appcenter-sdk-apple** version 3.3.3 and 3.3.4 carthage build failed with following:
```
*** Building scheme "AppCenterCrashes macOS Framework" in AppCenter.xcworkspace
Build Failed
	Task failed with exit code 65:
	/usr/bin/xcrun xcodebuild -workspace /Users/user/Carthage/Checkouts/appcenter-sdk-apple/AppCenter.xcworkspace -scheme AppCenterCrashes\ macOS\ Framework -configuration Release -derivedDataPath /var/root/Library/Caches/org.carthage.CarthageKit/DerivedData/12.0_12A7209/appcenter-sdk-apple/3.3.3 ONLY_ACTIVE_ARCH=NO CODE_SIGNING_REQUIRED=NO CODE_SIGN_IDENTITY= CARTHAGE=YES archive -archivePath /var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/appcenter-sdk-apple SKIP_INSTALL=YES GCC_INSTRUMENT_PROGRAM_FLOW_ARCS=NO CLANG_ENABLE_CODE_COVERAGE=NO STRIP_INSTALLED_PRODUCT=NO (launched in /Users/user/Carthage/Checkouts/appcenter-sdk-apple)
This usually indicates that project itself failed to compile. Please check the xcodebuild log for more details: /var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/carthage-xcodebuild.oCxAhA.log
```

With **appcenter-sdk-apple** version 3.3.1 and 3.3.2 carthage build failed with following:
```
*** Building scheme "AppCenter macOS Framework" in AppCenter.xcworkspace
Build Failed
	Task failed with exit code 65:
	/usr/bin/xcrun xcodebuild -workspace /Users/user/Carthage/Checkouts/appcenter-sdk-apple/AppCenter.xcworkspace -scheme AppCenter\ macOS\ Framework -configuration Release -derivedDataPath /var/root/Library/Caches/org.carthage.CarthageKit/DerivedData/12.0_12A7209/appcenter-sdk-apple/3.3.1 ONLY_ACTIVE_ARCH=NO CODE_SIGNING_REQUIRED=NO CODE_SIGN_IDENTITY= CARTHAGE=YES archive -archivePath /var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/appcenter-sdk-apple SKIP_INSTALL=YES GCC_INSTRUMENT_PROGRAM_FLOW_ARCS=NO CLANG_ENABLE_CODE_COVERAGE=NO STRIP_INSTALLED_PRODUCT=NO (launched in /Users/user/Carthage/Checkouts/appcenter-sdk-apple)
This usually indicates that project itself failed to compile. Please check the xcodebuild log for more details: /var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/carthage-xcodebuild.j3H0JU.log
```

Log files are missing at paths provided by error.

**Comment by Jamminroot** [comment]
Thanks for giving some insight. Please let us some time to investigate the problem and see if we can do anything about it.

**Comment by denis-ionov** [comment]
@Jamminroot log part with errors:
[err.log](https://github.com/microsoft/appcenter-sdk-apple/files/5274671/err.log)

**Comment by Jamminroot** [comment]
Tested scenario with 2 different macOS versions:
Test 1:
macOS: 10.15.6
Carthage: 0.36.0
Xcode: 12.0 (12A7209)

Test 2:

macOS: 11.0 Beta
Carthage: 0.36.0
Xcode: 12.0 (12A7209)

Both work as expected:

```
> XCODE_XCCONFIG_FILE=./Carthage.xcconfig carthage build --no-use-binaries --platform macOS
*** xcodebuild output can be found in /var/folders/7t/mgqf9w812zv7dqy7sswgw_fh0000gn/T/carthage-xcodebuild.TfDNPL.log
*** Building scheme "AppCenter macOS Framework" in AppCenter.xcworkspace
*** Building scheme "AppCenterAnalytics macOS Framework" in AppCenter.xcworkspace
*** Building scheme "AppCenterCrashes macOS Framework" in AppCenter.xcworkspace
*** Building scheme "AppCenterPush macOS Framework" in AppCenter.xcworkspace
```

Is can you check CLI tools version with `xcode-select --print-path`?

**Comment by denis-ionov** [comment]
```
xcode-select -p       
/Applications/Xcode.app/Contents/Developer

xcode-select -v
xcode-select version 2373.
```

**Comment by Jamminroot** [comment]
When investigating, we've missed the point with paths and completely missed the error clearly stated in logs you provided  🤦 My apologies.
That's something that needs to be addressed in PLCR. We'll address the issue.
Note, however, we do not officially support MacOS older than 10.14.4: https://docs.microsoft.com/en-us/appcenter/sdk/getting-started/macos#1-prerequisites

**Comment by denis-ionov** [comment]
Thanks! Waiting for feedback from PLCR 😊

**Comment by AnastasiaKubova** [comment]
The fix has been released in [4.0.0](https://github.com/microsoft/appcenter-sdk-apple/releases/tag/4.0.0) version of App Center.
